### PR TITLE
Add build-and-test file for Tioga

### PR DESCRIPTION
**Pull request by kab163**
RAJA would like to create a pipeline on Tioga - probably Umpire is to follow next with that. This sets up the baseline Tioga pipeline build-and-test file.

### CODE REVIEW

**Review by adrienbernede — CHANGES_REQUESTED** [review]
Thank you @kab163, this is looking like a good start.

In fact, I think this file looks good, but I would like to see it tested in the RAJA branch before we merge it: you can reference that branch directly in RAJA’s `.gitlab-ci.yml`. Now that you created this PR, the branch will exist on GitLab side, so RAJA can pick it up.

Also, could you please update `subscribed-pipelines.yml` and `custom-jobs-and-variables` in the `customization` folder? Those are templates for users not to start from scratch, and we should keep them as current as possible.

Thanks!

File: tioga-build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      #flux mini batch ${TIOGA_BUILD_AND_TEST_SHARED_ALLOC} --job-name=${ALLOC_NAME} --wrap sleep inf
+
+status_pending:
+  extends: [.on_tioga]
+  stage: allocate_resources
+  script:
+    - export pipeline_status="pending"
+    - *status_report
+
+rocm_5_1_1_clang_13_0_0:
```
Comment (rhornung67) [inline tioga-build-and-test.yml:76]: We should be testing 5.3.0 which is the system default version.

File: tioga-build-and-test.yml [hunk 2]
```diff
@@ -73,9 +73,9 @@ status_pending:
     - export pipeline_status="pending"
     - *status_report
 
-rocm_5_1_1_clang_13_0_0:
+rocm_5_3_0_clang_13_0_0:
```
Comment (rhornung67) [inline tioga-build-and-test.yml:76]: 👍

Comment (adrienbernede) [reply tioga-build-and-test.yml:76]: @kab163 you may have to create a PR in radiuss-spack-configs too to reference the appropriate rocm install.

Comment (kab163) [reply tioga-build-and-test.yml:76]: So given that I merged those changes to the radiuss-spack-configs, is there something needed here (in this repo) to pull in those changes? Or will it just be ok?

Comment (adrienbernede) [reply tioga-build-and-test.yml:76]: You only need to update the radiuss-spack-configs in the project repo (here RAJA).
 There is an implicit need for coherency between radiuss-spack-confg and radiuss shared CI.

**Comment by adrienbernede** [comment]
@kab163 I integrated your changes in #17

**Comment by adrienbernede** [comment]
Let’s use this branch to add changes if needed.

**Pull request merged**